CI: set xattr with empty value failed.

**What happened**:

https://github.com/juicedata/juicefs/actions/runs/14353104376/job/40236272349
should be introduced by https://github.com/juicedata/juicefs/pull/5938

```
============================= test session starts ==============================
platform linux -- Python 3.10.12, pytest-8.3.5, pluggy-1.5.0
rootdir: /home/runner/work/juicefs/juicefs/pyxattr
collected 287 items

tests/test_xattr.py FFFFFFFFFFFF........................................ [ 18%]
........................................................................ [ 43%]
........................................................................ [ 68%]
........................................................................ [ 93%]
...................                                                      [100%]

=================================== FAILURES ===================================
_________________________ test_empty_value[file name] __________________________

subject = ('/tmp/jfs/tmpw7kqu0vn/xattr-8fu8col0.test', False)

    def test_empty_value(subject):
        item, nofollow = subject
        xattr.set(item, USER_ATTR, EMPTY_VAL, nofollow=nofollow)
>       assert xattr.get(item, USER_ATTR, nofollow=nofollow) == EMPTY_VAL
E       OSError: [Errno [61](https://github.com/juicedata/juicefs/actions/runs/14353104376/job/40236272349#step:10:62)] No data available

tests/test_xattr.py:230: OSError
_____________________ test_empty_value[file name (bytes)] ______________________
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

--- a/pkg/meta/sql.go
+++ b/pkg/meta/sql.go
@@ -3527,10 +3527,7 @@ func (m *dbMeta) doSetXattr(ctx Context, inode Ino, name string, value []byte, f
 			}
 			_, err = s.Update(&x, k)
 		default:
-			if bytes.Equal(existing, value) {
-				return nil
-			}
-			if ok {
+			if ok && !bytes.Equal(existing, value) {
 				_, err = s.Update(&x, k)
 			} else {
 				err = mustInsert(s, &x)

--- a/pkg/meta/sql.go
+++ b/pkg/meta/sql.go
@@ -3527,10 +3527,10 @@ func (m *dbMeta) doSetXattr(ctx Context, inode Ino, name string, value []byte, f
 			}
 			_, err = s.Update(&x, k)
 		default:
-			if ok && !bytes.Equal(existing, value) {
-				_, err = s.Update(&x, k)
-			} else {
+			if !ok {
 				err = mustInsert(s, &x)
+			} else if !bytes.Equal(existing, value) {
+				_, err = s.Update(&x, k)
 			}
 		}
 		return err

--- a/pkg/meta/tkv.go
+++ b/pkg/meta/tkv.go
@@ -2683,7 +2683,7 @@ func (m *kvMeta) doSetXattr(ctx Context, inode Ino, name string, value []byte, f
 				return ENOATTR
 			}
 		}
-		if !bytes.Equal(v, value) {
+		if v == nil || !bytes.Equal(v, value) {
 			tx.set(key, value)
 		}
 		return nil

